src/libos/crates/async-io/src/util/channel.rs

```diff
@@ -280,9 +418,91 @@ impl Consumer {
         let rb = this_end.ringbuf();
         rb.len()
     }
+
+    pub fn shutdown(&self) {
+        self.common.consumer.shutdown()
+    }
+
+    pub fn is_shutdown(&self) -> bool {
+        self.common.consumer.is_shutdown()
+    }
+
+    pub fn status_flags(&self) -> StatusFlags {
+        self.this_end().flags.load(Ordering::Relaxed)
+    }
+
+    pub fn set_status_flags(&self, new_status: StatusFlags) -> Result<()> {
+        check_status_flags(new_status)?;
+        self.this_end().flags.store(new_status, Ordering::Relaxed);
+        Ok(())
+    }
+
+    pub fn is_nonblocking(&self) -> bool {
+        self.status_flags().contains(StatusFlags::O_NONBLOCK)
+    }
+
+    pub fn pollee(&self) -> &Pollee {
+        self.common.consumer.pollee()
+    }
+}
+
+impl Consumer<u8> {
+    pub async fn readv_async(&self, bufs: &mut [&mut [u8]]) -> Result<usize> {
+        let total_len: usize = bufs.iter().map(|buf| buf.len()).sum();
+        if total_len == 0 {
+            return Ok(0);
+        }
+
+        let mut poller = None;
+        loop {
+            // Attempt to read
+            let ret = self.try_readv(bufs);
+            if !ret.has_errno(EAGAIN) {
+                return ret;
+            }
+
+            if self.is_nonblocking() {
+                return_errno!(EAGAIN, "no data are present to be received");
+            }
+
+            if poller.is_none() {
+                poller = Some(Poller::new());
+            }
+            let mask = Events::IN;
+            let events = self.pollee().poll(mask, poller.as_mut());
+            if events.is_empty() {
+                poller.as_ref().unwrap().wait().await;
```

Comment:
We should use "poller.as_ref().unwrap().wait().await?" to handle the interrupt.

Reply:
Got it. Fixed.